src/cloudlabs/views_host.py

```diff
@@ -46,13 +48,82 @@ def add():
             fields['admin_ssh_key_id'] = form.admin_ssh_key.data
         else:
             fields['admin_password'] = form.admin_password.data
-        new_host = Host.create(**fields)
-        deploy(new_host)
-        flash('Host "{}" added'.format(form.label.data), 'success')
-        return redirect(url_for('main.index'))
+        if request.form.get('action', None) == 'Customise setup':
+            custom_form = CustomiseSetup()
+            db_host = Host(**fields)
+            custom_form.label = fields['label']
+            custom_form.dns_name = fields['dns_name']
+            custom_form.admin_username = fields['admin_username']
+            custom_form.git_repo = fields['git_repo']
+            custom_form.port = fields['port']
+            custom_form.setup_script.data = \
+                                         db_host.default_setup_script(**fields)
+            return render_template('setup_script.html', form=custom_form)
+
+        elif request.form.get('action', None) == 'Add':
+            new_host = Host.query.filter_by(label=fields['label'], 
```

Comment:
Remove this query and the if/else - here you just create a new host, no need to check, so it can match the original code path.